src/tree/tree_model.cc

```diff
@@ -618,6 +622,138 @@ std::string RegTree::DumpModel(const FeatureMap& fmap,
   return result;
 }
 
+class TreeReader : public JsonReader {
+  using Node = RegTree::Node;
+
+  void ForwardN(uint32_t n) {
+    for  (size_t i = 0 ; i < n; ++i) { cursor_.Forward(); }
+  }
+
+ public:
+  explicit TreeReader(StringView str) : JsonReader(str) {}
+
+  void ParseStat(int32_t n_nodes, std::vector<RTreeNodeStat>* p_stats) {
+    p_stats->resize(n_nodes);
+    auto& stats = *p_stats;
+    GetChar('[');
+    for (int32_t i = 0; i < n_nodes; ++i) {
+      GetChar('[');
+      auto& s = stats[i];
+      size_t pos {0};
+      s.loss_chg = std::stod(raw_str_.substr(cursor_.Pos(), 17), &pos);
```

Comment:
I cannot figure out why do we take 17 letters here. How do we know the size of the field?

Is it the size of serialized double?
What do you think about using std::numeric_limits<double>::max_digits10 ?

Reply:
`std::numeric_limits::max_digits10` is a good choice.  Again, thanks for the suggestion.